### PR TITLE
JAMES-2586 Postgres Subscription mapper

### PR DESCRIPTION
**Pull request by vttranlina**
jira https://issues.apache.org/jira/browse/JAMES-2586

### CODE REVIEW

File: backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresTableManager.java [hunk 1]
```diff
@@ -53,6 +55,26 @@ public Mono<Void> initializeTables() {
...
+
+    public Mono<Void> alterTableEnableRLS(PostgresTable table) {
+        return postgresExecutor.connection()
+            .flatMapMany(con -> con.createStatement(getAlterRLSStatement(table.getName())).execute())
+            .flatMap(Result::getRowsUpdated)
+            .then();
+    }
+
+    private String getAlterRLSStatement(String tableName) {
+        return "ALTER TABLE " + tableName + " ADD DOMAIN varchar(255) not null DEFAULT current_setting('app.current_domain')::text;" +
```
Comment (vttranlina) [inline backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresTableManager.java:73]: I can't find the API for doing that with Jooq, 
So I try to native SQL

Comment (quantranhong1999) [reply backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresTableManager.java:73]: I guess this should be done in https://github.com/linagora/james-project/issues/4898 instead.

Comment (vttranlina) [reply backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresTableManager.java:73]: I did cherrypick this commit to postgres table manager PR

**Comment by Arsnael** [comment]
Can be rebased

File: backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresDAO.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.postgres;
+
+import org.apache.james.backends.postgres.utils.PostgresExecutor;
+
+public abstract class PostgresDAO {
```
Comment (chibenwa) [inline backends-common/postgres/src/main/java/org/apache/james/backends/postgres/PostgresDAO.java:24]: I hardly see the use of this class.

If you want marking just use an interface?

File: mailbox/postgres/src/main/java/org/apache/james/mailbox/jpa/user/PostgresSubscriptionTable.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import org.apache.james.backends.postgres.PostgresIndex;
+import org.apache.james.backends.postgres.PostgresTable;
+import org.jooq.Field;
+import org.jooq.Record;
+import org.jooq.Table;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+
+public interface PostgresSubscriptionTable {
```
Comment (chibenwa) [inline mailbox/postgres/src/main/java/org/apache/james/mailbox/jpa/user/PostgresSubscriptionTable.java:30]: IMO move table definition into `PostgresSubscriptionModule`

File: mailbox/postgres/src/main/java/org/apache/james/mailbox/jpa/user/PostgresSubscriptionTable.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    Field<String> MAILBOX = DSL.field("mail_box", SQLDataType.VARCHAR(255).notNull());
```
Comment (chibenwa) [inline mailbox/postgres/src/main/java/org/apache/james/mailbox/jpa/user/PostgresSubscriptionTable.java:32]: mailbox

File: mailbox/postgres/src/main/java/org/apache/james/mailbox/jpa/user/PostgresSubscriptionTable.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    Field<String> USER = DSL.field("user_name", SQLDataType.VARCHAR(255).notNull());
+    Table<Record> TABLE_NAME = DSL.table("subscription");
+
+    PostgresTable TABLE = PostgresTable.name(TABLE_NAME.getName())
+        .createTableStep(((dsl, tableName) -> dsl.createTable(tableName)
+            .column(MAILBOX)
+            .column(USER)
+            .constraint(DSL.unique(MAILBOX, USER))))
+        .enableRLS();
```
Comment (chibenwa) [inline mailbox/postgres/src/main/java/org/apache/james/mailbox/jpa/user/PostgresSubscriptionTable.java:41]: enableRLS -> enableRowLevelSecurity

Please avoid acronyms they put an undue entry barrier to comprehension

**Review by chibenwa — COMMENTED** [review]
Overall +1

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**